src/Microsoft.Azure.WebJobs.Host/Bindings/BindingBase.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+
+    // Helper class for implementing IBinding with the attribute resolver pattern. 
```

Comment:
You call the pattern "attribute resolver" here, but the type is named AttributeCloner. Which is the better name do you think - AttributeResolver?